### PR TITLE
feat(17479): Edit MCDA functionality

### PR DESCRIPTION
**Pull request by albaranau**
https://kontur.fibery.io/Tasks/Task/Add-Edit-MCDA-flow-17479 

image

## Summary by CodeRabbit

- **New Features**
	- Introduced an `EditControl` component for editing MCDA configurations and layer states.
	- Added a `custom` property to MCDA configurations for custom settings.
	- Implemented new editing capabilities for layers, including a UI update with an edit cursor and color adjustments.
- **Enhancements**
	- Enhanced `useControlElements` to conditionally add an `EditControl` component based on layer editability.
	- Improved MCDA layer management with an `editMCDAConfig` function and updated `disableMCDALayer` to accept an `id` parameter.
	- Updated `MCDAForm` component with new form state management and a dynamic `disabled` state for a button based on data.
- **Refactor**
	- Refactored types and logic related to layer actions and states to support editability.
	- Modified `readMcdaJSONFile` to enhance the return object with a `custom` property.

### CODE REVIEW

File: src/features/mcda/components/MCDAForm/index.tsx [hunk 1]
```diff
@@ -14,7 +14,12 @@ import { availableBivariateAxisesAtom } from '../../atoms/availableBivariateAxis
 import s from './style.module.css';
 import type { Axis } from '~utils/bivariate';
 
-type FormState = {
+type FormInitialState = {
+  name: string;
+  axises: { id: string; label: string }[];
```
Comment (dqunbp) [inline src/features/mcda/components/MCDAForm/index.tsx:19]: axes

Comment (albaranau) [reply src/features/mcda/components/MCDAForm/index.tsx:19]: Yeah, I know it's axes in English, I decided to keep it consistent with the rest of the codebase, where this form ("axises") is used. That would be too much refactoring to change everywhere

File: src/core/logical_layers/utils/logicalLayerFabric.ts [hunk 2]
```diff
@@ -218,6 +223,17 @@ export function createLogicalLayerAtom(
         }
       });
 
+      onAction('edit', async () => {
+        try {
+          if (!state?.isEditable) return;
+          if (state.source?.style?.type === 'mcda') {
+          }
```
Comment (dqunbp) [inline src/core/logical_layers/utils/logicalLayerFabric.ts]: do we need empty if statement?

Comment (albaranau) [reply src/core/logical_layers/utils/logicalLayerFabric.ts]: It's not needed! And also the action handler it's in not needed too. I'm gonna clean this up, thanks!

Comment (albaranau) [reply src/core/logical_layers/utils/logicalLayerFabric.ts]: Removed along with unused action

**Review by dqunbp — APPROVED** [review]

File: src/features/mcda/components/MCDAForm/index.tsx (hunk 1 above)
Comment (Komzpa) [reply src/features/mcda/components/MCDAForm/index.tsx:19]: https://kontur.fibery.io/Tasks/Task/Change-typo-in-codebase-axises---axes-18017

Comment (albaranau) [reply src/features/mcda/components/MCDAForm/index.tsx:19]: @Komzpa thanks!

**Review by vkozio — APPROVED** [review]

File: src/features/mcda/mcdaConfig.tsx [hunk 3]
```diff
@@ -4,11 +4,43 @@ import { notificationServiceInstance } from '~core/notificationServiceInstance';
 import { i18n } from '~core/localization';
 import { formatBivariateAxisUnit, type Axis } from '~utils/bivariate';
 import { MCDAForm } from './components/MCDAForm';
+import type { LogicalLayerState } from '~core/logical_layers/types/logicalLayer';
 import type {
   MCDAConfig,
   MCDALayer,
 } from '~core/logical_layers/renderers/stylesConfigs/mcda/types';
 
+export async function editMCDAConfig(layerState: LogicalLayerState) {
```
Comment (amurKontur) [inline src/features/mcda/mcdaConfig.tsx:13]: maybe change the name of Create popup too? Update to Edit

**Pull request merged**